rcl/test/rcl/test_publisher.cpp

```diff
@@ -490,3 +535,291 @@ TEST_F(CLASSNAME(TestPublisherFixture, RMW_IMPLEMENTATION), test_invalid_publish
     rcl_publish_serialized_message(nullptr, &serialized_msg, nullptr));
   rcl_reset_error();
 }
+
+// Mocking rmw_publisher_count_matched_subscriptions to make
+// rcl_publisher_get_subscription_count fail
+TEST_F(
+  CLASSNAME(TestPublisherFixtureInit, RMW_IMPLEMENTATION),
+  test_mock_publisher_get_subscription_count)
+{
+  auto mock = mocking_utils::patch_and_return(
+    "lib:rcl", rmw_publisher_count_matched_subscriptions, RMW_RET_BAD_ALLOC);
+
+  // Now normal usage of the function rcl_publisher_get_subscription_count returning
+  // unexpected RMW_RET_BAD_ALLOC
+  size_t count_size;
```

Comment:
```suggestion
  size_t count_size = 0;
```
Also, can you check it's expected value here?

Reply:
See 5ed77be.

Reply:
See 5ed77be